packages/nimble-components/src/list-option/index.ts

```diff
@@ -51,6 +52,17 @@ export class ListOption extends FoundationListboxOption {
             .map(node => node.textContent?.trim())
             .join(' ');
     }
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        if (this.isDropdownOwner(this.parentElement)) {
+            this.parentElement.registerOption(this);
+        }
+    }
+
+    private isDropdownOwner(parent: unknown): parent is ListOptionOwner {
+        return typeof (parent as ListOptionOwner).registerOption === 'function';
```

Comment:
Some questions about types:
1. Why is `parent` of `unknown` type rather than `HTMLElement`? Isn't this only called with `this.parentElement: HTMLElement`?
1. Do we need to handle the case where `parent` is undefined? Or since this is called from connectedCallback we know it'll always be valid?

Reply:
1) I've updated the `ListOptionOwner` interface to extend `HTMLElement`, which allows me to get rid of the `unknown` typing.
2) I'm handling the case where `parent` is `null` (as `parentElement`'s type is `HTMLElement | null`.